README.md

```diff
@@ -0,0 +1,105 @@
+# Change-impact-analysis Tool
+## Table of Content
+- [Introduction](#introduction)
+- [Innovation](#innovation)
+- [How to Use](#how-to-use)
+- [Files Generated](#files-generated)
+- [Structure of the Tool](#structure-of-the-tool)
+  - [I. Compilation File List Generation](#i-compilation-file-list-generation)
+  - [II. Git Diff Report Generation](#ii-git-diff-report-generation)
+  - [III. Commit Metadata Retrieval](#iii-commit-metadata-retrieval)
+  - [IV. Web Script Generation](#iv-web-script-generation)
+
+## Introduction
+
+The Change Impact Analysis Tool generates a comprehensive visual report detailing changes in both header files and source code between two Linux versions (tags in the Linux kernel repository: old_tag and new_tag). This tool helps developers view updates from the old version.
+
+The diff report includes a subset of files from the Linux repository that are included in building the kernel, contributing to a focused and detailed report on the compile-time source code in Linux.
+
+## Innovation
+The idea of generating a web display for Linux kernel version change impact analysis is inspired by [Cregit](https://github.com/cregit/cregit). This tool innovates on Cregit by:
+
+- Considering the extensive code space the Linux kernel deals with, it provides a compile-time analysis instead of a static analysis of the commit history of the Linux kernel, presenting changes only in files used during compilation. 
+- Generating not only web source files but also lists of all source files and dependencies/header files used in kernel compilation, facilitating additional analysis purposes. (More details in [Files Generated](#files-generated))
+- Enabling comparison between two specific tags/releases in the Linux kernel, highlighting all newly added and deleted lines. This provides a clear layout of differences between the tags. While Cregit organizes information by files and embeds the latest commit details in each line/token, it does not support direct comparison between two tags.
+- User customization: allows users to define the URL of the Linux root repository and specify the subsystem for analysis. (More details in [How to Use](#how-to-use))
+
+## How to use
+
+To utilize this tool in your Linux environment (compatible with Ubuntu and Debian), follow these steps:
+
+Clone the repository:
+
+```bash
+git clone <repository_url>
+```
+
+Navigate to the cloned repository:
+
+```bash
+cd <repository_directory>
+```
+
+Execute the tool by specifying the old and new tags:
+
+```bash
+./run_tool.sh <tag1> <tag2> [-c clone_path] [-u repo_link] [-s subsystem]
```

Comment:
It might be good to provide an example with the values filled out here as well.

The reason this came to mind is that is not clear to me if `repo_link` should be a URL to a repo or a path to local repo. I think from the description below that should be a URL but that is not what I would have expected by default since the tool appears to rely on building first which would mean you need a local copy.

Reply:
Yeah! Addressed this issue with more detailed explanation and an example usage in the new commit.